src/main/java/org/influxdb/impl/GzipRequestInterceptor.java

```diff
@@ -0,0 +1,75 @@
+package org.influxdb.impl;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import okhttp3.Interceptor;
+import okhttp3.MediaType;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okio.BufferedSink;
+import okio.GzipSink;
+import okio.Okio;
+
+/**
+ * Implementation of a intercepter to compress http's body using GZIP.
+ *
+ * @author fujian1115 [at] gmail.com
+ */
+final class GzipRequestInterceptor implements Interceptor {
+
+    private AtomicBoolean enabled = new AtomicBoolean(false);
+
+    GzipRequestInterceptor() {
+    }
+
+    public void enable() {
+        enabled.set(true);
+    }
+
+    public boolean isEnabled() {
+        return enabled.get();
+    }
+
+    public void disable() {
+        enabled.set(false);
+    }
+
+    @Override
+    public Response intercept(Interceptor.Chain chain) throws IOException {
+        if (!enabled.get())
+            return chain.proceed(chain.request());
```

Comment:
@majst01  done and I squash commit to one to make commit clean. You can merge it now the close the issue #180 . Thanks again.